### PR TITLE
Fix kubernetes_with_qsfs script

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Issue happened because worker node was the same as the master's. Fixed by excluding the master node while filtering.

### Changes

Add types, remove unnecessary code, ping node before deploying on it

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3195

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts [hunk 1]
```diff
@@ -42,45 +42,56 @@ async function main() {
...
+
+    if (freeHRU >= disk_size && usedResources <= disk_size * count) {
+      qsfsNodes.push(node.nodeId);
+      usedResources += freeHRU;
+    }
+
+    if (usedResources >= disk_size * count) {
+      break;
+    }
+  }
```
Comment (AhmedHanafy725) [inline packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: why do we need that? the filter nodes method is already getting nodes with requested resources

File: packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts [hunk 2]
```diff
@@ -147,8 +158,8 @@ async function main() {
   await getDeployment(grid3, name);
 
   // //Uncomment the line below to cancel the deployment
-  // await cancel(grid3, { name: k.name });
-  // await deleteQsfs(grid3, { name: qsfs_name });
+  await cancel(grid3, k.name);
+  await deleteQsfs(grid3, qsfs_name);
```
Comment (AhmedHanafy725) [inline packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: please return these lines commented

File: packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts [hunk 3]
```diff
@@ -42,45 +42,56 @@ async function main() {
...
-  const workerQueryOptions: FilterOptions = {
-    cru: 1,
-    mru: 1, // GB
-    sru: 3,
-    availableFor: grid3.twinId,
-    farmId: 1,
-  };
-
   const qsfsQueryOptions: FilterOptions = {
     hru: 15,
```
Comment (AhmedHanafy725) [inline packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: why 15? count * disk_size

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Review by amiraabouhadid — DISMISSED** [review]
passes
image
but shouldn't it be ``freeHRU >= disk_size * count``?

File: packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts [hunk 4]
```diff
@@ -42,45 +42,55 @@ async function main() {
...
   const qsfsNodes: number[] = [];
+  let usedResources = 0;
+
+  const nodes = await grid3.capacity.filterNodes(qsfsQueryOptions);
+
+  // Loop over filtered nodes and push in qsfsNodes until they have the required resources
+  for (const node of nodes) {
+    const freeHRU = (node.total_resources.hru - node.used_resources.hru) / 1024 ** 3;
+
+    if (freeHRU >= disk_size) {
```
Comment (amiraabouhadid) [inline packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts:71]: shouldn't it be ```freeHRU >= disk_size * count ```?

Comment (amiraabouhadid) [inline packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: can you please explain the logic behind this conditional?

Comment (zaelgohary) [reply packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: If node's free hru is >= the disk size of the qsfs, then push the node to qsfs nodes array and accumulate usedResources.

Comment (amiraabouhadid) [reply packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: what is count?

Comment (zaelgohary) [reply packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts]: It's the number of qsfs deployments. So we need x deployments with the same exact resources mentioned in the qsfs options.

**Review by amiraabouhadid — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**